src/filtrering/filtrering-status/fargekategori.tsx

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import hiddenIf from '../../components/hidden-if/hidden-if';
+import {useStatustallVeilederSelector} from '../../hooks/redux/use-statustall';
+import './arbeidsliste.css';
+import {ReactComponent as ArbeidslisteikonBla} from '../../components/ikoner/arbeidsliste/arbeidslisteikon_bla.svg';
+import {ReactComponent as ArbeidslisteikonGronn} from '../../components/ikoner/arbeidsliste/arbeidslisteikon_gronn.svg';
+import {ReactComponent as ArbeidslisteikonLilla} from '../../components/ikoner/arbeidsliste/arbeidslisteikon_lilla.svg';
+import {ReactComponent as ArbeidslisteikonGul} from '../../components/ikoner/arbeidsliste/arbeidslisteikon_gul.svg';
+import BarInputCheckbox from '../../components/barinput/barinput-checkbox';
+import {
+    alleFargekategoriFilterAlternativer,
+    FARGEKATEGORI_A,
+    FARGEKATEGORI_B,
+    FARGEKATEGORI_C,
+    FARGEKATEGORI_D,
+    FARGEKATEGORI_E,
+    FARGEKATEGORI_F,
+    FARGEKATEGORI_INGEN_KATEGORI,
+    ferdigfilterListeLabelTekst,
+    MINE_FARGEKATEGORIER
+} from '../filter-konstanter';
+import {usePortefoljeSelector} from '../../hooks/redux/use-portefolje-selector';
+import {OversiktType} from '../../ducks/ui/listevisning';
+import {useDispatch} from 'react-redux';
+import {FARGEKATEGORIER_HOVEDFILTER_KLIKK, FARGEKATEGORIER_UNDERFILTER_KLIKK} from '../../ducks/filtrering';
+import {FargekategoriModell} from '../../model-interfaces';
+
+function FilterStatusMineFargekategorier() {
+    const dispatch = useDispatch();
+    const statusTall = useStatustallVeilederSelector();
+    const filtervalg = usePortefoljeSelector(OversiktType.minOversikt).filtervalg;
+    const ferdigfilter = filtervalg.ferdigfilterListe;
+    const fargekategoriFilter = filtervalg.fargekategorier;
+
+    const hovedfilterChecked =
+        ferdigfilter.includes(MINE_FARGEKATEGORIER) &&
+        alleFargekategoriFilterAlternativer.every(f => fargekategoriFilter.includes(f));
+    const hovedfilterIndeterminate =
```

Comment:
Need bit more explanation on this line

Reply:
I you look at the image i pasted in the other comment, the "Hovedfilter" can have multiple states; `unchecked`, `checked`, `indeterminate`. The "Underfilter" affects which state the "Hovedfilter" is in.